src/icmp.rs

```diff
@@ -1,43 +1,1572 @@
 use core::mem;
+use core::net;
 
-pub const ICMP_HDR_LEN: usize = mem::size_of::<IcmpHdr>();
+/// An enum representing either an ICMPv4 or ICMPv6 header.
+///
+/// - `V4` contains an IPv4 ICMP header as defined in RFC 792 (see `IcmpHdr`)
+/// - `V6` contains an IPv6 ICMP header as defined in RFC 4443 (see `IcmpV6Hdr`)
+///
+/// This enum allows working with both ICMP protocol versions through a unified interface.
+pub enum Icmp {
+    V4(IcmpHdr),
+    V6(IcmpV6Hdr),
+}
 
-#[repr(C)]
+/// Represents an ICMP header as defined in RFC 792.
+/// The header consists of a type and code field identifying the message type,
+/// a checksum for error detection, and a data field whose format depends on the message type.
+///
+/// The `type_` field identifies the general category of message, such as:
+/// - 0: Echo Reply
+/// - 3: Destination Unreachable
+/// - 5: Redirect
+/// - 8: Echo Request
+/// - 30: Traceroute
+/// - 40: PHOTURIS
+///
+/// The `code` field provides additional context for the message type.
+///
+/// The `check` field contains a checksum calculated over the ICMP header and its payload.
+///
+/// The `data` field contains type-specific data such as echo identifiers/sequence numbers,
+/// redirect gateway addresses, or pointers to errors in received packets.
+#[repr(C, packed)]
 #[derive(Copy, Clone)]
 #[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
 pub struct IcmpHdr {
     pub type_: u8,
     pub code: u8,
-    pub checksum: u16,
-    pub un: IcmpHdrUn,
+    pub check: [u8; 2],
```

Comment:
the original name was more explicit imo, would you mind change it to `checksum` ?

Reply:
I agree with the option 1.

What's odd is that the types in Linux kernel aren't consistent, some of them using `check`, some of them `checksum`:

https://elixir.bootlin.com/linux/v6.15/source/include/uapi/linux/ip.h#L103
https://elixir.bootlin.com/linux/v6.15/source/include/uapi/linux/icmp.h#L92

But since both we and Linux use shortened names like `src`, `addr`, I think `check` makes sense too.

Reply:
I've [updated the README.md](https://github.com/vadorovsky/network-types/pull/48/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R112) to include that callout.